lib/clone.js

```diff
@@ -49,21 +52,24 @@ const cloneTask = (sshGitHubUrl, repoName, progressBar) => {
 const clone = (services, task = cloneTask) => {
   process.stdout.write(`Cloning repos...${newline}`)
 
-  const gitHubDetails = getGitHubDetails(services)
+  const gitHubRepoInfo = getGitHubDetails(services)
   const progressBar = new Progress(':percent [:bar] (:current/:total)', {
-    total: gitHubDetails.length,
+    total: gitHubRepoInfo.length,
     clear: true
   })
+  const githubInfoParentArray = chunkArray(gitHubRepoInfo, numberOfCPUs * 4)
```

Comment:
Yeah it feels like this might be unnecessarily punishing those who didn't see the problem in the first place. I didn't look to much into it yet, but something like tracking the memory needed by each clone, averaging it, checking against `freemem`, then polling it for when there's enough to kick off another.

As I say, it's just an idea but working with what's available over determining a ceiling would optimise to the machine's capabilities.

With the proposed approach, someone with a less powerful machine might hit the same problem and we'd just keep reducing that number until it runs at a crawl for all.